stl/inc/variant

```diff
@@ -1506,39 +1521,49 @@ struct _Variant_all_visit_results_same<_Callable, _Meta_list<_IndexVectors...>,
                                           // sequences of indices in _IndexVectors has the same type and value category.
 };
 
+template <class _To, class _Callable, class _ListOfIndexVectors, class... _Variants>
+struct _Variant_all_visit_results_implicitly_convertible; // undefined
+
+template <class _To, class _Callable, class... _IndexVectors, class... _Variants>
+struct _Variant_all_visit_results_implicitly_convertible<_To, _Callable, _Meta_list<_IndexVectors...>, _Variants...>
+    : _All_convertible<_To, typename _Variant_single_visit_result<_Callable, _IndexVectors,
+                                _Variants...>::type...>::type { // true_type if invocation of _Callable on the elements
+                                                                // of _Variants with all sequences of indices in
+                                                                // _IndexVectors is implicitly convertible to _To.
+};
+
 template <int _Strategy>
 struct _Visit_strategy;
 
 template <>
 struct _Visit_strategy<-1> { // Fallback strategy for visitations with too many total states for the
                              // following "switch" strategies.
-    template <class _ListOfIndexVectors, class _Callable, class... _Variants>
-    static constexpr _Variant_visit_result_t<_Callable, _Variants...> _Visit(
+    template <class _Ret, class _ListOfIndexVectors, class _Callable, class... _Variants>
+    static constexpr _Ret _Visit(
         size_t _Idx, _Callable&& _Obj, _Variants&&... _Args) { // dispatch a visitation with many potential states
         constexpr size_t _Size = _Variant_total_states<_Remove_cvref_t<_Variants>...>;
         static_assert(_Size > 256);
         constexpr auto& _Array =
-            _Variant_dispatch_table<_ListOfIndexVectors, _Callable, _Meta_list<_Variants...>>::_Array;
+            _Variant_dispatch_table<_Ret, _ListOfIndexVectors, _Callable, _Meta_list<_Variants...>>::_Array;
         return _Array[_Idx](static_cast<_Callable&&>(_Obj), static_cast<_Variants&&>(_Args)...);
     }
 };
 
 template <>
 struct _Visit_strategy<0> {
-    template <class, class _Callable>
-    static constexpr _Variant_visit_result_t<_Callable> _Visit(
-        size_t, _Callable&& _Obj) { // dispatch a visitation with 4^0 potential states
+    template <class _Ret, class, class _Callable>
+    static constexpr _Ret _Visit(size_t, _Callable&& _Obj) { // dispatch a visitation with 4^0 potential states
         return static_cast<_Callable&&>(_Obj)();

```

Comment:
I think this needs something like:
```c++
if constexpr(is_void_v<_Ret>) {
    return static_cast<void>(static_cast<_Callable&&>(_Obj)());
} else {
    return static_cast<_Callable&&>(_Obj)();
}
```
so silly uses like `std::visit<void>([] { return 42; });` work.